packages/new-editor/src/components/editor/variables/master/Variables.tsx

```diff
@@ -54,31 +54,29 @@ export const Variables = ({ variables, setVariables, setSelectedVariablePath }:
   });
 
   const addVariable = () => {
-    const newVariables = structuredClone(variables);
     const newVariable: Variable = {
       name: '',
       value: '',
       description: '',
-      metadata: 'none',
+      metadata: { type: '' },
       children: []
     };
 
-    const addChildToFirstSelectedRowReturnValue = addChildToFirstSelectedRow(table, newVariables, newVariable);
+    const addChildToFirstSelectedRowReturnValue = addChildToFirstSelectedRow(table, variables, newVariable);
     const parentNode = addChildToFirstSelectedRowReturnValue.selectedNode;
     if (parentNode) {
       parentNode.value = '';
-      parentNode.metadata = 'none';
+      parentNode.metadata = { type: '' };
     }
 
     setSelectedVariablePath(addChildToFirstSelectedRowReturnValue.newChildPath);
-    setVariables(newVariables);
+    setVariables(addChildToFirstSelectedRowReturnValue.newData);
   };
 
   const deleteVariable = () => {
-    const newVariables = structuredClone(variables);
-    const selectedVariablePath = deleteFirstSelectedRow(table, newVariables);
-    setSelectedVariablePath(selectedVariablePath);
-    setVariables(newVariables);
+    const deleteFirstSelectedRowReturnValue = deleteFirstSelectedRow(table, variables);
+    setSelectedVariablePath(deleteFirstSelectedRowReturnValue.selectedVariablePath);
+    setVariables(deleteFirstSelectedRowReturnValue.newData);
```

Comment:
same here as above -> rule of react, do not modify state directly. 
structured clone would "saved" you from this. if you don't want to use it, use the lambda method from setVariable.

Reply:
I moved the structured cloning inside of the methods that actually perform changes to the state. This way, I don't need to pass a clone to these methods every time and I cannot forget to do it in the first place. I also test that these methods do not perform any changes to the data passed to them. See:

https://github.com/axonivy/config-editor-client/blob/c6e8b8fb0a86ab6ae57e44f8c142d2b5b7159b04/packages/new-editor/src/data/data.ts#L28-L36

and

https://github.com/axonivy/config-editor-client/blob/c6e8b8fb0a86ab6ae57e44f8c142d2b5b7159b04/packages/new-editor/src/data/data.test.ts#L75-L92

In this case, `deleteFirstSelectedRow` even returns an object with additional information. So, a switch to the lambda method from `setVariables` wouldn't be that straight forward.

Reply:
Ah okey 👍 
Ok for me :)